app/models/posting_item.rb

```diff
@@ -32,4 +32,9 @@
 
 class PostingItem < ApplicationRecord
     belongs_to :posting
+    belongs_to :commodity_category
+
+    def stock_balance
+    	PostingItem.sum(:quantity).where(account_id: 7).group(:commodity_category_id)
```

Comment:
The account_id shouldn't be hard coded. We can use something like this: "Account.find_by({'code': :stock})"

Reply:
I agree. Not supposed to be pushed in the PR. It was an experimentation code. Removed!